app/services/plans/update_service.rb

```diff
@@ -197,5 +199,16 @@ def discard_charge!(charge)
 
       Invoice.where(id: draft_invoice_ids).update_all(ready_to_be_refreshed: true) # rubocop:disable Rails/SkipsModelValidations
     end
+
+    # NOTE: We should remove pending subscriptions
+    #       if plan has been downgraded but amount cents became less than downgraded value. This pending subscription
+    #       is not relevant in this case and downgrade should be ignored
+    def process_downgraded_subscriptions
+      return unless plan.subscriptions.active.exists?
+
+      Subscription.where(previous_subscription_id: plan.subscriptions.active, status: :pending).find_each do |sub|
+        sub.destroy! if plan.amount_cents < sub.plan.amount_cents
```

Comment:
Any reason for destroying rather than canceling?

Reply:
Yep, using cancel status now, wasn't aware of it and it wasn't part of specs:)